### ISSUE TITLE
Does Automation/Select overlaps work for you?

### ISSUE THREAD

**Issue by slrslr** [TypesettingTools/Aegisub#197]
Open .srt file, menu "Automation" > Select overlaps
cause
`Lua reported a runtime error` on Linux, Aegisub 3.2.2
[Detail](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1005171)
Is anyone capable to fix it, i am willing to pay for that fix. Thank you

The file:
https://github.com/TypesettingTools/Aegisub/blob/master/automation/autoload/select-overlaps.moon
(in /usr/share/aegisub/automation/autoload/select-overlaps.moon) 

ChatGPT suggested this modification:

Replace:
`[prepare i, line for i, line in ipairs subs when is_dialogue line]`

by:
`[prepare i, line for i, line in ipairs(subs) when is_dialogue(line)]`

or by:
`[prepare(i, line) for i, line in ipairs(subs) when is_dialogue(line)]`

saying:

> "The issue in the provided Lua script is with the ipairs function being used on the subs object. The error message indicates that Lua is expecting a table but received a userdata instead."

But it does not work. Do you have please idea what to try? I am trying to detect and fix subtitle overlaps in several .ass files. 

An issue of the luajit and the current Aegisub version?

**Comment by arch1t3cht** [comment]
This is an issue with Debian's Aegisub package. Aegisub needs luajit to be compiled with the Lua 5.2 compat flag, which Debian's luajit does not have. This results in the `__ipairs` metatable entry on the `subs` object not being respected.

To fix this you would need to compile Aegisub yourself with `--force-fallback-for=luajit`.

**Comment by aniolm9** [comment]
Sorry for commenting after the bug was closed, but I'm the Debian Maintainer for Aegisub and I want to clarify this issue because I keep getting e-mails about this.
As I explained some years ago in https://github.com/Aegisub/Aegisub/pull/141, the Debian version of luajit is not compiled with the Lua 5.2 compat flag, so it can't be used with Aegisub.
On the other hand, uploading a new package version to the current stable is not straightforward (see https://www.debian.org/doc/manuals/developers-reference/pkgs.html#upload-stable), so the changes I can make are minimal. What I can do is trying to build version 3.2.2 with Lua 5.2 and without luajit. If this works, I think that it could get into stable.
First I'll upload 3.4.0 to unstable (probably today or tomorrow) and then I'll try to update 3.2.2.
